packages/ember-data/lib/system/model/model.js

```diff
@@ -375,7 +374,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
   */
   toJSON: function(options) {
     // container is for lazy transform lookups
-    var serializer = JSONSerializer.create({ container: this.container });
+    var serializer = this.store.serializerFor('-default');
```

Comment:
You are right, thanks! This part of the code is old, I see now that the issue I worked around has been solved for ~2 years :smile: